tests/e2e/happy_path_test.go

```diff
@@ -0,0 +1,299 @@
+	"io"
+	"net/http"
+	"os/exec"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	gatewayapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+
+	mcpv1alpha1 "github.com/kagenti/mcp-gateway/pkg/apis/mcp/v1alpha1"
+)
+
+var _ = Describe("MCP Gateway Happy Path", func() {
+	var (
+		httpRoute1 *gatewayapiv1.HTTPRoute
+		httpRoute2 *gatewayapiv1.HTTPRoute
+		mcpServer1 *mcpv1alpha1.MCPServer
+		mcpServer2 *mcpv1alpha1.MCPServer
+	)
+
+	BeforeEach(func() {
+		// create httproutes for test servers that should already be deployed
+		httpRoute1 = BuildTestHTTPRoute("e2e-server1-route", TestNamespace,
+			"server1.mcp.example.com", "mcp-test-server1", 9090)
+		httpRoute2 = BuildTestHTTPRoute("e2e-server2-route", TestNamespace,
+			"server2.mcp.example.com", "mcp-test-server2", 9090)
+
+		// create mcpserver resources
+		mcpServer1 = BuildTestMCPServer("e2e-mcpserver1", TestNamespace,
+			"e2e-server1-route", "s1_")
+		mcpServer2 = BuildTestMCPServer("e2e-mcpserver2", TestNamespace,
+			"e2e-server2-route", "s2_")
+	})
+
+	AfterEach(func() {
+		// cleanup in reverse order
+		if mcpServer1 != nil {
+			CleanupResource(ctx, k8sClient, mcpServer1)
+		}
+		if mcpServer2 != nil {
+			CleanupResource(ctx, k8sClient, mcpServer2)
+		}
+		if httpRoute1 != nil {
+			CleanupResource(ctx, k8sClient, httpRoute1)
+		}
+		if httpRoute2 != nil {
+			CleanupResource(ctx, k8sClient, httpRoute2)
+		}
+	})
+
+	It("should aggregate MCP servers and manage HTTPRoute conditions", func() {
+		By("Creating HTTPRoutes")
+		Expect(k8sClient.Create(ctx, httpRoute1)).To(Succeed())
+		Expect(k8sClient.Create(ctx, httpRoute2)).To(Succeed())
+
+		By("Creating MCPServer resources")
+		Expect(k8sClient.Create(ctx, mcpServer1)).To(Succeed())
+		Expect(k8sClient.Create(ctx, mcpServer2)).To(Succeed())
+
+		By("Verifying MCPServers become ready")
+		VerifyMCPServerReady(ctx, k8sClient, mcpServer1.Name, mcpServer1.Namespace)
+		VerifyMCPServerReady(ctx, k8sClient, mcpServer2.Name, mcpServer2.Namespace)
+
+		By("Verifying ConfigMap is created with correct content")
+		VerifyConfigMapExists(ctx, k8sClient)
+
+		configMap := &corev1.ConfigMap{}
+		Expect(k8sClient.Get(ctx, types.NamespacedName{
+			Name:      ConfigMapName,
+			Namespace: SystemNamespace,
+		}, configMap)).To(Succeed())
+
+		// parse and verify configmap content
+		configContent := configMap.Data["config.yaml"]
+		fmt.Printf("ConfigMap content:\n%s\n", configContent)
+		Expect(configContent).To(ContainSubstring("s1_"))
+		Expect(configContent).To(ContainSubstring("s2_"))
+		Expect(configContent).To(ContainSubstring("mcp-test-server1.mcp-test.svc.cluster.local"))
+		Expect(configContent).To(ContainSubstring("mcp-test-server2.mcp-test.svc.cluster.local"))
+
+		By("Waiting for broker to register servers")
+		// wait for broker to load the config and be ready
+		// we'll verify by checking if the broker pod has reloaded config
+		Eventually(func() bool {
```

Comment:
Yeah I can see that this might be an issue as it is eventually consistent but might take longer than the given time. I guess this is where the api to post a config update to comes from.

Reply:
Yeah, https://github.com/kagenti/mcp-gateway/pull/146 makes this much more predictable (and fast)